website/www/site/content/en/blog/dyi-content-discovery-platform-genai-beam.md

```diff
@@ -0,0 +1,329 @@
+---
+layout: post
+title:  "DIY GenAI Content Discovery Platform with Apache Beam"
+date:   2023-09-27 00:00:01 -0800
+categories:
+  - blog
+authors:
+  - prodriguezdefino
+  - namita sharma
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# DIY GenAI Content Discovery Platform with Apache Beam
+
+## Introduction
+
+Your digital assets, such as documents, PDFs, spreadsheets, and presentations, contain a wealth of valuable information which sometimes is hard to find or discover. This blog post discusses how to build a DIY starter architecture, based on near real time ingestion processing and large language models (LLMs) to extract meaningful information from your assets, making it available and discoverable through a simple natural language query.
+
+Building a near real time processing pipeline for content ingestion may seem like a very complex task, and it can be. To alliviate this task Apache Beam framework exposes a set of powerful constructs to remove the complexities of interacting with a variety of content sources and destinations, error handling, clarity, modularity, all while maintaining a good degree of resiliency and scalability with minimal effort. An Apache Beam streaming pipeline should be able to connect to the many components of a solution to quickly process document's content ingestion requests, making the information available in few seconds after the ingestion request was triggered.
```

Comment:
@prodriguezdefino - Looks like there are some typos. (e.g. alliviate instead of alleviate). Could you fix any typos in the doc please?

Reply:
done